### ISSUE TITLE
Microsoft.Cdn/profiles/endpoints/customdomains

### ISSUE THREAD

**Issue by shenglol** [Azure/arm-template-whatif#140]
## Describe the noise

**Resource type** (i.e. Microsoft.Storage/storageAccounts)
Microsoft.Cdn/profiles/endpoints/customdomains

**apiVersion** (i.e. 2019-04-01)
2019-12-31

**Client (PowerShell, Azure CLI, or API)**
PowerShell

**Relevant ARM Template code (we only need the resource object for the above `resourceType` and `apiVersion`, but if it's easier you can include the entire template**

**Expected response (i.e. "I expected no noise since the template has not been modified since the resources were deployed)**
No change.

**Current (noisy) response (either include a screenshot of the what-if output, or copy/paste the text)**
```
  ~ Microsoft.Cdn/profiles/thetradedesk-ads-html5-cdn/endpoints/thetradedesk-ads-html5 [2019-12-31]
    - properties.customDomains: [
        0:

          name:                "azhtml5-adsrvr-org"
          properties.hostName: "azhtml5.adsrvr.org"

      ]

  ~ Microsoft.Cdn/profiles/thetradedesk-ads-html5-cdn/endpoints/thetradedesk-ads-html5/customdomains/azhtml5-adsrvr-org [2019-12-31]
    - properties.customHttpsParameters:

        certificateSource:                             "AzureKeyVault"
        certificateSourceParameters.@odata.type:       "#Microsoft.Azure.Cdn.Models.KeyVaultCertificateSourceParameters"
        certificateSourceParameters.deleteRule:        "NoAction"
        certificateSourceParameters.resourceGroupName: "va9-ttd-tls-certs-rg"
        certificateSourceParameters.secretName:        "STARadsrvrorg"
        certificateSourceParameters.secretVersion:     "cb20f743b7f8497ba762bd9fb33fc9d2"
        certificateSourceParameters.subscriptionId:    "001a3882-eb1c-42ac-9edc-5e2872a07783"
        certificateSourceParameters.updateRule:        "NoAction"
        certificateSourceParameters.vaultName:         "va9-TTDImportedCerts-kv"
        minimumTlsVersion:                             "TLS12"
        protocolType:                                  "ServerNameIndication"

```

**Additional context**

**Comment by alex-frankel** [comment]
@detienne20 is this w51?

**Comment by detienne20** [comment]
w51

**Comment by alex-frankel** [comment]
This fix should be fully deployed. Please let us know if you are still seeing the issue.

**Comment by alex-frankel** [comment]
Re-opening as it looks like some of the noise is still present:

```
  ~ Microsoft.Cdn/profiles/thetradedesk-ads-html5-cdn [2019-12-31]
    - kind: "cdn"

  ~ Microsoft.Cdn/profiles/thetradedesk-ads-html5-cdn/endpoints/thetradedesk-ads-html5 [2019-12-31]
    - properties.customDomains: [
        0:

          name:                "azhtml5-adsrvr-org"
          properties.hostName: "azhtml5.adsrvr.org"

      ]

```

**Comment by shenglol** [comment]
The following property is covered by the Swagger round trip validation KPI:
- $.kind (2021-02-06)

**Comment by kleky** [comment]
Hi, we're having this same issue too. All of a sudden the what-if is identifying these changes to custom domain, but also to Azure for MySql admin password

```
 - properties.customHttpsParameters:

        certificateSource:                           "Cdn"
        certificateSourceParameters.@odata.type:     "#Microsoft.Azure.Cdn.Models.CdnCertificateSourceParameters"
        certificateSourceParameters.certificateType: "Dedicated"
        minimumTlsVersion:                           "TLS12"
        protocolType:                                "ServerNameIndication"
```

```
~ Microsoft.DBforMySQL/servers/rch*****dev [2017-12-01]
    + properties.administratorLoginPassword: "*******"
    + properties.createMode:                 "Default"
```

**Comment by kleky** [comment]
Is this a what-if bug, so when I deploy these changes "shouldn't" be applied, given that nothing has changed since the last deployment?

**Comment by alex-frankel** [comment]
Hi @kleky - these are what-if noise issues. The properties under `properties.customHttpsParameters` look like default values. The `DBforMySql/servers` noise should be filed as a separate issue, but those look like write-only properties that are not tagged as such in the `DBforMySql/servers` api definition.

**Comment by torreymicrosoft** [comment]
Thanks for your patience, this was recently picked up as one of the older open issues for the ARM Deployments team to revisit, and apologies for the silence since 2021. This one was reopened back then because the first fix did not fully land, and it then went quiet, which is not a good look on a five year old issue.

What-If for Azure Deployment Stacks went generally available today, and noise reduction is enabled in all regions. A stack records a what-if baseline when it is deployed, and any property that is unchanged between that baseline and the current evaluation is removed from the result as a false positive.

All three properties on this thread are the right shape for it, and I checked each against the current type definitions with Bicep 0.46.1 rather than trusting the old `fix committed` label:

- `kind` on `Microsoft.Cdn/profiles` did not exist in the type definition at all at your 2019-12-31 (BCP187), and is explicitly read-only at the current 2025-06-01 (BCP073).
- `properties.customDomains` on `Microsoft.Cdn/profiles/endpoints` is read-only at 2025-06-01 (BCP073).
- `properties.customHttpsParameters` on `Microsoft.Cdn/profiles/endpoints/customDomains` is read-only at 2025-06-01 (BCP073).

In other words all three are values the CDN service owns and returns on GET, none of them can ever appear in your template, so what-if has always had to report them as removals. Constant and repeatable on every run is exactly the input the baseline is built to absorb.

You reported this from PowerShell, so at resource group scope:

```
New-AzResourceGroupDeploymentStackWhatIfResult `
  -Name whatif-preview `
  -ResourceGroupName <resource-group> `
  -StackResourceId <deployment-stack-resource-id> `
  -TemplateFile main.bicep `
  -ActionOnUnmanage DetachAll `
  -DenySettingsMode None `
  -RetentionInterval PT3H
```

One honest caveat before you try it. The baseline is recorded when the stack is deployed or updated, so deploy or update the stack once to establish it, then run what-if against the stack. The docs are also clear that noise reduction removes many of these differences but not every one, so it is still worth reading the result rather than assuming everything left is real.

Announcement: https://techcommunity.microsoft.com/blog/azuregovernanceandmanagementblog/now-generally-available-what-if-for-azure-deployment-stacks/4547614
Docs: https://learn.microsoft.com/azure/azure-resource-manager/bicep/deployment-stacks-what-if

One loose end from the thread: the `Microsoft.DBforMySQL/servers` `properties.administratorLoginPassword` line @kleky raised in 2021 is a different problem and should not be filed here. A write-only credential showing with a `+` is a security concern rather than cosmetic noise, so if you still see it, please raise it separately rather than as noise.

I am closing this out. If `kind`, `customDomains` or `customHttpsParameters` still show up in a Stacks What-if result once the baseline is in place, reopen here or file in `Azure/deployment-stacks` with the what-if result resource ID.
